src/MOI/MathOptInterface.jl

```diff
@@ -0,0 +1,100 @@
+using MathOptInterface
+
+const MOI = MathOptInterface
+
+export CplexSolver
+
+struct CplexSolver <: MOI.AbstractSolver
+    mipstart_effortlevel::Cint
+    options
+end
+function CplexSolver(;mipstart_effortlevel::Cint = CPX_MIPSTART_AUTO, options...)
+    CplexSolver(mipstart_effortlevel, options)
+end
+
+
+const Linear = MOI.ScalarAffineFunction{Float64}
+const LE = MOI.LessThan{Float64}
+const GE = MOI.GreaterThan{Float64}
+const EQ = MOI.EqualTo{Float64}
+const LinConstrRef{T} = MOI.ConstraintReference{Linear, T}
+const SVConstrRef{T} = MOI.ConstraintReference{MOI.SingleVariable, T}
+
+const SUPPORTED_OBJECTIVES = [
+    Linear
+    # ScalarQuadraticFunction{Float64}
+]
+
+const SUPPORTED_CONSTRAINTS = [
+    (Linear, EQ),
+    (Linear, LE),
+    (Linear, GE),
+    (MOI.SingleVariable, EQ),
+    (MOI.SingleVariable, LE),
+    (MOI.SingleVariable, GE),
+    (MOI.SingleVariable, MOI.Interval{Float64})
+]
+
+function MOI.supportsproblem(s::CplexSolver, objective_type, constraint_types)
+    if !(objective_type in SUPPORTED_OBJECTIVES)
+        return false
+    end
+    for c in constraint_types
+        if !(c in SUPPORTED_CONSTRAINTS)
+            return false
+        end
+    end
+    return true
+end
+
+struct ConstraintMapping
+    # rows in constraint matrix
+    less_than::Dict{LinConstrRef{LE}, Int}
+    greater_than::Dict{LinConstrRef{GE}, Int}
+    equal_to::Dict{LinConstrRef{EQ}, Int}
+
+    # references to variable
+    variable_upper_bound::Dict{SVConstrRef{LE}, MOI.VariableReference}
```

Comment:
I was thinking that solvers could skip these mappings by matching up the constraint reference number with the variable reference number, since multiple bounds per variable are not allowed.